src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -582,5 +569,114 @@ private ReadGroupCountColumns(final ExomeReadCounts tool) {
+
+        /**
+         * Creates a new group-by option.
+         *
+         * @param countColumnsFactory the count-columns manager factory to be used with
+         *                            this group-by option. Assumed not to be {@code null}.
+         */
+        GroupBy(final Function<ExomeReadCounts, ? extends CountColumns> countColumnsFactory) {
+            this.countColumnsFactory = countColumnsFactory;
+        }
+
+        /**
+         * Returns a lowercase user friendly name for this group-by option.
+         * <p>
+         * <p>This is to be used for some user targeted messages (error, log, etc)</p>
+         *
+         * @return never {@code null}.
+         */
+        @Override
+        public String toString() {
+            return name().toLowerCase().replace("_", " ");
+        }
+
+        /**
+         * Creates a column count manager for a given read-counts tool.
+         *
+         * @param tool the targeted tool.
+         * @return never {@code null}.
+         * @throws IllegalArgumentException if {@code tool} is {@code null}.
+         * @throws IllegalStateException    if {@code tool} is not in a compatible state for this type of count-columns.
+         */
+        protected CountColumns countColumns(final ExomeReadCounts tool) {
+            return countColumnsFactory.apply(tool);
+        }
+    }
+
+    /**
+     * Matrix value transformation function to apply to the main output matrix values.
+     */
+    protected enum Transform {
+
+        /**
+         * Default raw value (non-)transformation.
+         */
+        RAW((count, columnTotal) -> Integer.toString(count)),
+
+        /**
+         * Proportional coverage transformation.
+         */
+        PCOV((count, columnTotal) ->
+                String.format(PCOV_OUTPUT_DOUBLE_FORMAT, count / (double) columnTotal));
+
+        @FunctionalInterface
+        protected interface Operator {
+            String apply(final int count, final long columnTotal);
+        }
+
+        /**
+         * Output matrix value transformer.
+         * <p>
+         * <p>It takes exactly two double arguments: the output matrix value and the column sum total.</p>
+         */
+        protected final Operator operator;
+
+        /**
+         * Creates a Transform instance given the corresponding operator and format.
+         *
+         * @param operator the value transformation operator.
+         */
+        Transform(final Operator operator) {
+            this.operator = operator;
+        }
+
+        /**
+         * Transforms and composes the string representation of the output value.
+         *
+         * @param value       the output matrix value.
+         * @param columnTotal the corresponding column total.
+         * @return never {@code null}.
+         */
+        public String apply(final int value, final long columnTotal) {
+            return operator.apply(value, columnTotal);
```

Comment:
The total can actually be 0. In this case PCOV would result in NaN values which I believe is a reasonable output in this case. Thus value >= 0, value <= total are enough.